packages/flutter_tools/lib/src/windows/build_windows.dart

```diff
@@ -19,12 +19,25 @@ import '../globals.dart' as globals;
 import '../migrations/cmake_custom_command_migration.dart';
 import 'visual_studio.dart';
 
+// These characters appear to be fine: @%()-+_{}[]`~
+const String badCharacters = r"'#!$^&*=|,;<>?";
+
 /// Builds the Windows project using msbuild.
 Future<void> buildWindows(WindowsProject windowsProject, BuildInfo buildInfo, {
   String? target,
   VisualStudio? visualStudioOverride,
   SizeAnalyzer? sizeAnalyzer,
 }) async {
+  final String projectPath = windowsProject.parent.directory.absolute.path;
```

Comment:
In the future, it'll be difficult to remember *why* this check exists without looking at the commit history. I would add a quick comment about how CMake generates MSBuild files that does not properly escape the paths. Consider also adding a link to the issue.